numpy/lib/histograms.py

```diff
@@ -349,6 +378,44 @@ def _unsigned_subtract(a, b):
         # signed to unsigned
         return np.subtract(a, b, casting='unsafe', dtype=dt)
 
+def _get_auto_bin_edges(a, n_equal_bins, first_edge, last_edge):
+    """
+    Computes the bins used internally by `np.histogram(bins='auto')`.
+
+    Parameters
+    ==========
+    a : ndarray
+        Ravelled data array
+    first_edge, last_edge
+        Forwarded arguments from `_get_outer_edges`.
+    n_equal_bins : 
+        Forwarded arugment from _get_bin_edges,
+        this function only called when n_equal_bins > 2
+
+    Returns
+    =======
+    bin_edges : ndarray
+        Array of bin edges
+    """
+    keep = (a >= first_edge)
+    keep &= (a <= last_edge)
+    if not np.logical_and.reduce(keep):
+        a = a[keep]
+
+    if a.size == 0:
+        return np.array([0,1])
+
+    width = np.ptp((first_edge,last_edge))/(n_equal_bins)
+    zeroed_min = a - first_edge
+    quantized = np.floor_divide(zeroed_min, width) * width
+    left_edges = np.unique(quantized)
+    right_edges = left_edges + width
+    unzeroed_edges = np.unique((left_edges, right_edges)) + first_edge
+    edges = np.concatenate(([first_edge], unzeroed_edges, [last_edge]))
+
+    TOL = width/2
+    widths = np.append(width, np.diff(edges)) #remove bins produced by float errors
```

Comment:
`append` is needless indirection here
```suggestion
    widths = np.concatenate(([width], np.diff(edges)))  # remove bins produced by float errors
```